src/hooks/executor_start.c

```diff
@@ -0,0 +1,76 @@
+#include <postgres.h>
+
+#include "executor_start.h"
+
+#include <executor/executor.h>
+#include <nodes/pg_list.h>
+#include <nodes/plannodes.h>
+
+#include "plan_tree_walker.h"
+#include "utils.h"
+
+typedef struct
+{
+    List *oidList;
+    bool  isIndexScan;
+} OperatorUsedCorrectlyContext;
+
+static bool operator_used_correctly_walker(Node *node, OperatorUsedCorrectlyContext *context)
+{
+    if(node == NULL) return false;
+    if(IsA(node, IndexScan)) {
+        context->isIndexScan = true;
+        bool status = plan_tree_walker(node, operator_used_correctly_walker, (void *)context);
+        context->isIndexScan = false;
+        return status;
+    }
+    if(IsA(node, OpExpr)) {
+        OpExpr *opExpr = (OpExpr *)node;
+        if(list_member_oid(context->oidList, opExpr->opno) && !context->isIndexScan) {
+            return true;
```

Comment:
Does this function return 'true' when the operator is used correctly or when the operator is used incorrectly?

Reply:
Used incorrectly

Reply:
Renamed for clarity.